src/App/ResetCryptoStore.ts

```diff
@@ -0,0 +1,89 @@
+    const registration = await parseRegistrationFile(registrationFile);
+    Logger.configure({
+        console: config.logging.level,
+        colorize: config.logging.colorize,
+        json: config.logging.json,
+        timestampFormat: config.logging.timestampFormat
+    });
+    LogService.setLogger(Logger.botSdkLogger);
+
+    const {appservice, storage, cryptoStorage} = getAppservice(config, registration);
+    if (!cryptoStorage) {
+        log.info(`Encryption is not enabled in the configuration file "${configFile}", so there is no encryption state to be reset`);
+        return;
+    }
+
+    const botUsersManager = new BotUsersManager(config, appservice);
+
+    for (const botUser of botUsersManager.botUsers) {
+        try {
+            const userStorage = storage.storageForUser?.(botUser.userId)
+            if (!userStorage) {
+                log.warn(`No storage for ${botUser.userId}`);
+                continue;
+            }
+            const accessToken = await Promise.resolve(userStorage?.readValue("accessToken"));
+            if (!accessToken) {
+                log.debug(`No access token for ${botUser.userId}: no session to remove`);
+                continue;
+            }
+
+            const userCryptoStorage = cryptoStorage?.storageForUser(botUser.userId);
+            if (!userCryptoStorage) {
+                log.warn(`No crypto storage for ${botUser.userId}`);
+                continue;
+            }
+            const deviceId = await userCryptoStorage?.getDeviceId();
+            if (!deviceId) {
+                log.debug(`No crypto device ID for ${botUser.userId}: no crypto state to remove`);
+                continue;
+            }
+
+            const client = new MatrixClient(config.bridge.url, accessToken, userStorage, userCryptoStorage);
+            await client.doRequest("POST", "/_matrix/client/v3/logout", {
+                user_id: botUser.userId,
+                "org.matrix.msc3202.device_id": deviceId,
+            });
+            log.info(`Logged out crypto device for ${botUser.userId}`)
+
+            try {
+                await Promise.resolve(userStorage.storeValue("accessToken", ""))
+                log.info(`Deleted access token for ${botUser.userId}`)
+            } catch (err: unknown) {
+                log.error(`Failed to delete access token for ${botUser.userId}: ${err}`)
+            }
+
+        } catch (err: unknown) {
+            log.error(`Failed to log out crypto device for ${botUser.userId}: ${err}`)
+        }
+    }
+
+    if (config.encryption?.storagePath) {
+        try {
+            await rm(config.encryption.storagePath, { recursive: true, force: true });
+            log.info("Removed crypto store from disk")
+        } catch (err) {
+            log.error("Failed to remove crypto store from disk")
+        }
+    }
+    // Process hangs without this, maybe because of a leftover promise
```

Comment:
*Suspicious noises*

Reply:
Your suggestion to use [why-is-node-still-running](https://www.npmjs.com/package/why-is-node-still-running) did reveal the cause of the hang after all!: the Redis store.  Closing it lets the process terminate cleanly.